code/game/objects/items/devices/scanners.dm

```diff
@@ -328,6 +328,11 @@ SLIME SCANNER
 		if(cyberimp_detect)
 			to_chat(user, "<span class='notice'>Detected cybernetic modifications:</span>")
 			to_chat(user, "<span class='notice'>[cyberimp_detect]</span>")
+	var/list/nanite_data = list()
+	SEND_SIGNAL(M, COMSIG_NANITE_GET_DATA, nanite_data)
+	if(LAZYLEN(nanite_data) && !nanite_data["stealth"])
+		to_chat(user, "<span class='notice'><b>Nanites Detected</b></span>")
+		to_chat(user, "<span class='notice'>Saturation: [nanite_data["nanite_volume"]]/[nanite_data["max_nanites"]]</span>")

```

Comment:
You're doing this backwards, the component should be detecting when a healthscan happens and say text to the user.